ui/packages/components/src/DatePicker/TimeInput.tsx

```diff
@@ -0,0 +1,190 @@
+import { useEffect, useState } from 'react';
+import { format } from 'date-fns';
+import { ZodError, type ZodSchema } from 'zod';
+
+import { cn } from '../utils/classNames';
+import {
+  getHourSchema,
+  millisecondsSchema,
+  minutesSchema,
+  periodSchema,
+  secondsSchema,
+} from './timeSchemas';
+
+type HandleTimeChangeProps = {
+  e: React.ChangeEvent<HTMLInputElement>;
+  schema: ZodSchema<any>;
+  setInputValue: React.Dispatch<React.SetStateAction<string>>;
+};
+
+type TimeInputProps = {
+  is24HourFormat?: boolean;
+  selectedTime?: Date;
+  onSelect: React.Dispatch<React.SetStateAction<Date | undefined>>;
+  setIsValidTime: React.Dispatch<React.SetStateAction<boolean>>;
+  isValidTime: boolean;
+};
+
+export function TimeInput({
+  selectedTime,
+  onSelect,
+  is24HourFormat = false,
+  setIsValidTime,
+  isValidTime,
+}: TimeInputProps) {
+  const [hourInput, setHourInput] = useState(
+    selectedTime ? format(selectedTime, is24HourFormat ? 'HH' : 'hh') : ''
+  );
+  const [minuteInput, setMinuteInput] = useState(selectedTime ? format(selectedTime, 'mm') : '');
+  const [secondInput, setSecondInput] = useState(selectedTime ? format(selectedTime, 'ss') : '');
+  const [millisecondInput, setMillisecondInput] = useState(
+    selectedTime ? format(selectedTime, 'SSS') : ''
+  );
+  const [periodInput, setPeriodInput] = useState(
+    selectedTime && !is24HourFormat ? format(selectedTime, 'a') : ''
+  );
```

Comment:
After playing around with the date picker, I thought it'd be nice to default the time values to 0s. This allows me to leave the minutes/seconds/milliseconds/period alone if I don't care about them

```suggestion
  const [hourInput, setHourInput] = useState(
    selectedTime ? format(selectedTime, is24HourFormat ? 'HH' : 'hh') : '00'
  );
  const [minuteInput, setMinuteInput] = useState(selectedTime ? format(selectedTime, 'mm') : '00');
  const [secondInput, setSecondInput] = useState(selectedTime ? format(selectedTime, 'ss') : '00');
  const [millisecondInput, setMillisecondInput] = useState(
    selectedTime ? format(selectedTime, 'SSS') : '000'
  );
  const [periodInput, setPeriodInput] = useState(
    selectedTime && !is24HourFormat ? format(selectedTime, 'a') : 'AM'
  );
```